src/motion/functionality/types.ts

```diff
@@ -10,15 +10,23 @@ export interface FunctionalProps extends MotionProps {
 }
 
 export interface FunctionalComponentDefinition {
+    key: string
     shouldRender: (props: MotionProps) => boolean
-    component: ComponentType<FunctionalProps>
+    Component: ComponentType<FunctionalProps>
 }
 
-export type UseFunctionalityComponents<P = {}> = (
-    props: P & MotionProps,
+export type LoadFunctionalityComponents<P = {}> = (
+    ref: RefObject<Element>,
     values: MotionValuesMap,
+    props: P & MotionProps,
     controls: ValueAnimationControls<P>,
+    inherit: boolean
+) => ReactElement<FunctionalProps>[]
+
+export type RenderComponent<P = {}> = (
     ref: RefObject<Element>,
     style: CSSProperties,
+    values: MotionValuesMap,
+    props: P,
     isStatic?: boolean
-) => ReactElement<P>[]
+) => ReactElement
```

Comment:
why was the generic type removed here?

Reply:
Basically it wasn't correct and started flagging.